source/Stargate-Examples-Tests/SouthAmericanCurrenciesRESTfulControllerTest.class.st

```diff
@@ -49,6 +62,35 @@ SouthAmericanCurrenciesRESTfulControllerTest >> testCantGetArgentinePesoAskingFo
 			]
 ]
 
+{ #category : #tests }
+SouthAmericanCurrenciesRESTfulControllerTest >> testCantGetArgentinePesoAskingForUnsupportedLanguage [
+
+	| request unsupportedLanguage |
+	unsupportedLanguage := 'de'.
+
+	request := self
+		           requestToGETResourceIdentifiedBy: 'ARS'
+		           acceptingContentMatching:
+		           resourceController currencyVersion1dot0dot0MediaType
+		           inAnyOf: unsupportedLanguage.
+
+	self
+		should: [ 
+			resourceController
+				currencyBasedOn: request
+				within: self newHttpRequestContext ]
+		raise: HTTPNotAcceptable
+		withExceptionDo: [ :error | 
+			self
+				assert: error messageText
+				equals:
+					'Cannot generate a response entity with acceptable characteristics.';
+				assert: error code equals: 406;
+				assert: error allowedLanguageTags size equals: 2;
+				assert: error allowedLanguageTags includes: 'en-US' asLanguageTag;
```

Comment:
Do the extract method for these other language ranges. I already gave you name suggestions